daemon/container_operations_windows.go

```diff
@@ -163,7 +163,13 @@ func serviceDiscoveryOnDefaultNetwork() bool {
 	return true
 }
 
-func setupPathsAndSandboxOptions(container *container.Container, cfg *config.Config, sboxOptions *[]libnetwork.SandboxOption) error {
+func buildSandboxPlatformOptions(container *container.Container, cfg *config.Config, sboxOptions *[]libnetwork.SandboxOption) error {
+	// By default, the Windows internal resolver does not forward requests to
+	// external resolvers - but forwarding can be enabled using feature flag
+	// "windows-dns-proxy":true.
+	if doproxy, exists := cfg.Features["windows-dns-proxy"]; !exists || !doproxy {
```

Comment:
Wondering if we should leave the old name `windows-dns-no-proxy` and turn the conditional here into `exists || !doproxy`.

The reason why I think it could be better is that it's a bit confusing to set feature to `false` to opt-in into a new behavior and IMO it's more intuitive if the missing value is equivalent to explicit `false` rather than `true`.

The alternative would be to change the `windows-dns-proxy` into `windows-dns-no-proxy` later, but that would mean that users that opt-in now, will have to adjust their config again.

WDYT?

Reply:
I changed it because it got a bit double-negative-y, I think this naming is a bit simpler. Disabling it, if needed, in 27.0 will look like this - I think its meaning is clear? ...

```
  "features": {
	  "windows-dns-proxy": false
  },
```

Docs also have an example of an enabled-by-default feature that can be disabled ... https://docs.docker.com/reference/cli/dockerd/#feature-options 

So, I prefer it this way - but can change it back if needed?

Reply:
(Either way - I'll raise a new PR to change the default, and mark it for milestone 27.0, as soon as this one's merged.)

Reply:
No strong opinion on my side, but maybe @thaJeztah has some preference?

Reply:
Hm... yeah, the ones where defaults change are always a bit tricky I guess. So the new default will become "disabled by default" (but currently it's "enabled by default"?)

From some perspective `windows-dns-no-proxy` or `windows-disable-dns-proxy`  or `windows-dns-proxy-disabled` would be more friendly, but also considering what it would look like if we would be showing the features as part of `docker info`


I don't think we currently expose the `features`, but I think that's something we could consider, and if we do, for those to reflect the defaults (in this case, that would mean `windows-dns-proxy: false`?)

Reply:
Yes, for release 26.1.0 the default will be `windows-dns-proxy: false`.

In 27.0.0, we'll change the default to `windows-dns-proxy: true`.

If those were displayed in `docker info`, that's what they'd look like ... clear, I think?

(I reckon our minds are corrupted by Go's zero-values ... we think a default needs to be `false`, that's why I originally used `windows-dns-no-proxy`. But, for normal people (!), I think getting rid of the double-negative in `windows-dns-no-proxy: false` is a good thing.)

Reply:
FWIW, I think getting rid of the double-negative is great. IMHO it feels more natural to have to turn a feature-flag 'true' to get it enabled and 'false' to opt-out.